compare_platforms/compare_platforms_read_only.c

```diff
@@ -0,0 +1,209 @@
+        assert((publicreview + bigsz) <=  (max(realname, permissions)));
+        return 1; 
+    }
+}
+
+void free_pointers(struct review *rv)
+{
+    free(rv->realname);
+    free(rv->publicreview);
+    free(rv->permissions);
+}
+
+void rewrite_permissions_by_overflow(struct review *rv, size_t smsz, size_t bigsz, bool bWeak)
+{
+    int revpos = assert_no_overlap(rv->realname, rv->publicreview, rv->permissions, smsz, bigsz);
+    // realname has to come before permissions for the following to make sense
+    assert((rv->realname + smsz) <= rv->permissions);
+        
+    if((revpos == 1) && (rv->publicreview+bigsz <= rv->permissions)) 
+    {
+        // if publicreview is in the middle and permissions is on the right of it
+        assert((rv->publicreview + bigsz + bigsz/2) > rv->permissions);
+        printf("\nOverflowing publicreview into permissions by 1:\n");
+        overflow_into_permissions(bigsz+1, 1, rv->publicreview, rv);
+    }
+    else 
+    {
+        assert((rv->realname + smsz + smsz/2) > rv->permissions);
+        printf("\nOverflowing realname into permissions by 1:\n");
+        overflow_into_permissions(smsz+1, 1, rv->realname, rv);
+    }
+    
+    const size_t oversz = rv->permissions - rv->realname + 2 - smsz;
+    printf("\nOverflowing realname into permissions by %zu:\n", oversz);
+    overflow_into_permissions(smsz+2, oversz, rv->realname, rv);
+}
+
+int main(int argc, char* argv[])
+{
+    struct review review;
+    // In a non-CHERI environment, write protection is weakly enforced
+    bool bWeak = true;
+    
+    // initialization values for struct member sizes
+    const size_t smallsz = 0x20;
+    const size_t biggersz = 0x1001;
+       
+    char *realname = malloc(smallsz);
+    strcpy(realname, "Baba Yaga");
+    review.realname = realname;
+    
+    char *publicreview = malloc(biggersz);
+    strcpy(publicreview, "I am a little unclear as to the contribution. I think the authors could strengthen their case considerably if they conducted an RCT. Weak reject.");
+    review.publicreview = publicreview;
+    
+    char *permissions = malloc(smallsz);
+    strcpy(permissions, "r");
+    review.permissions = permissions;
+    
+    print_details(&review);
+    
+    bool b_improved = false;
+    
+    char *newpublicreview = malloc(biggersz);
+    strcpy(newpublicreview, "5 ***** strong accept. This author deserves two Nobels and an ice cream.\n");
+
+#ifdef __CHERI_PURE_CAPABILITY__    
+    if(can_write(&review) == false) 
+    {
+        struct review *ro_review = set_read_only(&review);
+        assert((cheri_perms_get(ro_review) & CHERI_PERM_STORE) == 0);
+
+        if ((argc > 1) && (strcmp(argv[1], "-overwrite") == 0)) 
+        {
+            printf("\nThe struct is read-only so trying to change the review will make the program crash.\n");
+            bWeak = false;
+            b_improved = change_publicreview(ro_review, newpublicreview, bWeak);
+            // This line should be unreachable.
```

Comment:
If a line should be unreachable, it would be best to make it `assert(false)` or some other failure state to ensure that.

Reply:
Fixed in d032455c2ca8a57911b597af27fb082fab8f2fd1